libraries/AP_SmartAudio/AP_SmartAudio.cpp

```diff
@@ -1047,9 +1054,9 @@ bool AP_SmartAudio::hot_deploy(){
 }
 
 // return true if power need to bee synced with the AP_VideoTx backend
-bool AP_SmartAudio::sync_power(){
-    if(_vtx_current_state->version==SMARTAUDIO_SPEC_PROTOCOL_v21){
-        return AP::vtx().get_power_mw()!=_get_power_in_mw_from_dbm(_vtx_current_state->power);
+bool AP_SmartAudio::sync_power() {
+    if(_vtx_current_state->version == SMARTAUDIO_SPEC_PROTOCOL_v21){
+        return AP::vtx().get_configured_power_dbm() != _vtx_current_state->power;
```

Comment:
When checking if sync is needed with the vtx backend, only configured power in params is checked, if the current_power_value differs from the configured one, then no update is performed.

Reply:
This is what AP_VideoTX::update_power() is supposed to tell you. My purpose here was to simplify the logic which uses two non obvious functions that are only used here - sometimes doing the obvious thing is more understandable :)

Reply:
Message received.